### PR TITLE
Topic controller's AdminClient config #307

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change
- Enhancement 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by sknot-rh — APPROVED** [review]

**Review by ppatierno — COMMENTED** [review]
Not sure we would need an env var for configuring the client-id from the outside.
It could be useful when you need to run more TO instances maybe related to the same Kafka cluster? Something that doesn't sound to make much sense. I would say that a kind of "hard-coded" client-id containing at least the name of the corresponding Kafka cluster (if possible) could be enough. But @tombentley could have a different opinion.

File: topic-operator/src/main/java/io/strimzi/operator/topic/Config.java [hunk 1]
```diff
@@ -119,6 +120,9 @@ private Value(String key, Type<? extends T> type, boolean required) {
     /** The kubernetes namespace in which to operate. */
     public static final Value<String> NAMESPACE = new Value<>(TC_NAMESPACE, STRING, true);
 
+    /** Cluster-ID */
+    public static final Value<String> CLIENT_ID = new Value<>(TC_CLIENT_ID, STRING, "");
```
Comment (tombentley) [inline topic-operator/src/main/java/io/strimzi/operator/topic/Config.java:124]: I think the default should be something more like `"strimzi-topic-operator-" + uuid`, that way it's at least vaguely useful even when not configured.

**Comment by tombentley** [comment]
> Not sure we would need an env var for configuring the client-id from the outside.
It could be useful when you need to run more TO instances maybe related to the same Kafka cluster? Something that doesn't sound to make much sense. I would say that a kind of "hard-coded" client-id containing at least the name of the corresponding Kafka cluster (if possible) could be enough. But @tombentley could have a different opinion.

I think there's two things here: 

1. Being able to specify a particular client id, which is what's implemented here and makes sense in standalone TO use cases.
2. Having a good client id in the CO-deploys-TO scenario. We can build on this PR to implement that. Possibly we should expose it through the API, but I think having the CO configure a good default (e.g. using the namespace and cluster name) should be done even if we don't expose it though the API.

**Comment by ppatierno** [comment]
Yeah I always forget that standalone TO is a kind of thing we still support :-)

File: topic-operator/src/main/java/io/strimzi/operator/topic/Config.java (hunk 1 above)
Comment (ShubhamRwt) [reply topic-operator/src/main/java/io/strimzi/operator/topic/Config.java:124]: Ok sure@tombentley , I will make this change. Regarding the UUID, can we use UUID.randomUUID() to generate it ?

Comment (tombentley) [reply topic-operator/src/main/java/io/strimzi/operator/topic/Config.java:124]: `UUID.randomUUID()` is fine with me.

Comment (ShubhamRwt) [reply topic-operator/src/main/java/io/strimzi/operator/topic/Config.java:124]: Okay. Thanks

Comment (ShubhamRwt) [reply topic-operator/src/main/java/io/strimzi/operator/topic/Config.java]: Hello @tombentley,  I have made the necessary changes as per mentioned above but I have not changed the tests. Do I need to change them or are they fine ?

**Comment by ShubhamRwt** [comment]
Hello @ppatierno , @tombentley , @sknot-rh , are the above  changes good to go or do we require some more changes?

**Review by tombentley — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**